version: "3.9"
services:
  postgres:
    image: postgres:15
    env_file: .env
    volumes:
      - "postgres-data:/var/lib/postgresql/data/"

  django:
    build: .
    env_file: .env
    depends_on:
      - postgres
    environment:
      - "DJANGO_ALLOWED_HOSTS=127.0.0.1,localhost"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`127.0.0.1`) || Host(`localhost`)"
      - "traefik.http.routers.django.entrypoints=web"

  reverse-proxy:
    image: traefik:v2.9
    env_file: .env
    command:
      #- "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

volumes:
  postgres-data:
